src/oscontainer.py

```diff
@@ -117,6 +118,14 @@ def oscontainer_build(containers_storage, src, ref, image_name_and_tag,
         print("Copying ostree commit into container: {} ...".format(rev))
         run_verbose(["ostree", "--repo=" + dest_repo, "pull-local", src, rev])
 
+        # Generate pkglist.txt in to the oscontainer at /
+        pkg_list_dest = os.path.join(mnt, 'pkglist.txt')
+        pkgs = RpmOstree.db_query_all(r, rev, None)
+        with open(pkg_list_dest, 'w') as f:
+            for pkg in pkgs:
```

Comment:
I think we may want `sorted(pkgs)` here.

Reply:
Although not sure that will work (probably not).  Maybe:
```
pkgs = [pkg.get_nevra() for pkg in pkgs]
pkgs.sort()
``` ?

Reply:
Although, it *might* be sorted internally by rpm-ostree, I'm getting sorted order out, but that might be a consequence of using the commit metadata which is definitely sorted already as opposed to querying the rpmdb.

Reply:
I think you're right that `sorted(pkgs)` won't work since I'm not sure how it'd know how to use e.g. the `rpm_ostree_package_cmp` to compare them.

Trying:

```
In [7]: pkgs = RpmOstree.db_query_all(r, "fedora/x86_64/coreos/testing-devel", None)

In [8]: len(pkgs)
Out[8]: 340

In [9]: new_pkgs = sorted(pkgs)
```

Hmm, it *didn't* error out. So somehow it worked?
Except:

```
In [12]: pkgs[0].get_nevra()
Out[12]: 'GeoIP-1.6.12-5.fc30.x86_64'

In [13]: new_pkgs[0].get_nevra()
Out[13]: 'bind-export-libs-32:9.11.6-5.P1.fc30.x86_64'

In [14]: new_pkgs[1].get_nevra()
Out[14]: 'GeoIP-1.6.12-5.fc30.x86_64'
```

I think it sorted it in some way, but not really the way we wanted to (maybe by pointer?).

Reply:
https://github.com/coreos/coreos-assembler/pull/664